src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb

```diff
@@ -173,7 +173,9 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
                 Dim iBaseAddress As UInteger
 
-                If (TypeOf (value) Is UInteger) Then
+                Dim throwAwayObject As Decimal = Nothing
+                If (TypeOf (value) Is UInteger OrElse
+                    (CpsPropertyDescriptorWrapper.IsAnyCpsComponent(m_Objects) AndAlso Decimal.TryParse(DirectCast(value, String), throwAwayObject))) Then
                     iBaseAddress = CUInt(value)
```

Comment:
@srivatsn is correct. Right now dynamic type system is always strings, because it is for UI and also msbuild properties are always strings. However, the various implementations of `object IProperty.GetValueAsync()` does try to return the value as the correct type. In this piece of code we are commenting on, what is the CPS object you are working on? Is it already the value from the `DynamicType` or something else?